domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -2986,15 +3086,15 @@ public Future<RowSet<Row>> select(String sql) {
   }
 
   /**
-   * Run a select query.
+   * Run a select query using the readonly connection.
    *
    * <p>To update see {@link #execute(String, Handler)}.
    *  @param sql - the sql query to run
    * @param queryTimeout query timeout in milliseconds, or 0 for no timeout
    * @param replyHandler the query result or the failure
    */
   public void select(String sql, int queryTimeout, Handler<AsyncResult<RowSet<Row>>> replyHandler) {
-    getSQLConnection(queryTimeout,
+    getSQLReadConnection(queryTimeout,
```

Comment:
sql can be a writing SQL: `UPDATE foo SET jsonb=$1 RETURNING jsonb`

Therefore this change cannot be made in this `select` method but requires a new `selectRead` method.

Reply:
> Therefore this change cannot be made in this select method but requires a new selectRead method.

A new selectRead method to be created that has the same arguments and return type as this select method?

Reply:
Yes.